### PR TITLE
install CAF(ignoring NO_PUBKEY error), install build-essential dependency

### PR DESCRIPTION
**Pull request by adityaketkar**
install CAF(ignoring NO_PUBKEY error), install build-essential dependency, and some minor fixes

### CODE REVIEW

**Review by prasadtalasila — COMMENTED** [review]
Please spin up a vagrant machine, run tool-install.sh and then run the following commands.    
```shell
> cd BITS-Darshini/data/packet
> /usr/bin/time bro -r packet_data.pcap
```

Do post the results of the command.

**Comment by adityaketkar** [comment]
The terminal shows this:

0.14user 0.01system 0:00.26elapsed 61%CPU (0avgtext+0avgdata 45320maxresident)k
240inputs+72outputs (1major+8296minor)pagefaults 0swaps

Had to do it without vagrant though. I have listed below some problems with vagrant by running vagrant based installation on different machines. We should try to fix them.

Problems with Vagrant based installation:-
1. It assumes the user has installed packages npm, mvn and VirtualBox.
               I think it should be mentioned in documentation to install VirtualBox and vagrant beforehand, 
                along with the steps to do it
              As for npm and mvn, we can include them as a dependency in install.sh.

2. Running vagrant makes the PC download "jdk-8u144-linux-x64.tar.gz"  all over again every time, even if the machine is configured with java-jdk beforehand. It takes unnecessary time to download a 180 MB file.

**Comment by prasadtalasila** [comment]
The tomcat7 throws a visible error, but gets installed successfully on vagrant. All vagrant setups are supposed to be clean startup. That's why the repeated java8 installations.

With the proposed tool-install.sh, only the source files of bro gets copied. The binaries do not exist.    

On the machine you are trying, you may be using the bro that got installed using apt-get. You need to check in the context of vagrant for a confirmation. Let me see if there is an easier way to do this and I will get back to you.

File: scripts/tool-install.sh [hunk 1]
```diff
@@ -12,5 +12,19 @@ sudo apt-get install pfring nprobe ntopng ntopng-data n2disk cento nbox
 sudo apt-get install wireshark tshark
 sudo apt-get update
 
+#install dependencies for bro
+sudo apt-get install cmake make gcc g++ flex bison libpcap-dev libssl-dev python-dev swig zlib1g-dev
+
+#install C++ Actors Framework, Ignore the KEYEXPIRED 1503492954 or NO_PUBKEY error
+sudo sh -c "echo 'deb http://download.opensuse.org/repositories/devel:/libraries:/caf/xUbuntu_14.04/ /' > /etc/apt/sources.list.d/caf.list"
+sudo apt-get update
+sudo apt-get install -y caf
```
Comment (prasadtalasila) [inline scripts/tool-install.sh]: working command here is:
```shell
sudo apt-get install -y caf --allow-unauthenticated
```

File: scripts/tool-install.sh [hunk 2]
```diff
@@ -12,5 +12,19 @@ sudo apt-get install pfring nprobe ntopng ntopng-data n2disk cento nbox
...
+
 #install bro
-sudo apt-get install bro
+sudo wget https://www.bro.org/downloads/bro-2.5.1.tar.gz
```
Comment (prasadtalasila) [inline scripts/tool-install.sh]: The installation commands given below do not install bro tool. Please see the [installation steps](https://www.digitalocean.com/community/tutorials/how-to-install-bro-on-ubuntu-16-04) and add the commands, url reference at the end of the script as comment.

**Review by prasadtalasila — CHANGES_REQUESTED** [review]
please make the suggested changes.

**Comment by adityaketkar** [comment]
I did the requested changes and redeployed the analyzer. Though the installation goes well without any errors, now the application gives blank screen as shown.


image

**Comment by prasadtalasila** [comment]
Are you logged into the cyberoam? I think there are a few css and js libraries that are pulled from WWW. Please check this again.

**Comment by adityaketkar** [comment]
Rebuilding the application solved the issue. "Populate Table" gives proper output when I followed along the  [Usage Tutorial](https://github.com/prasadtalasila/BITS-Darshini/wiki/Usage-Tutorial).

image

**Comment by prasadtalasila** [comment]
The changes proposed in this PR have been indirectly merged in the commit 315de3a. Closing this PR now.

**Comment by prasadtalasila** [comment]
Thanks @adityaketkar for the work.

**Pull request closed without merging**